src/libasr/pass/intrinsic_function_registry.h

```diff
@@ -1160,6 +1162,53 @@ static inline void verify_args(const ASR::IntrinsicScalarFunction_t& x,
 
 } // namespace BinaryIntrinsicFunction
 
+
+namespace ObjectType {
+
+     static inline void verify_args(const ASR::IntrinsicScalarFunction_t& x, diag::Diagnostics& diagnostics) {
+        ASRUtils::require_impl(x.n_args != 1,
+            "ASR Verify: type() takes only 1 argument `object`",
+            x.base.base.loc, diagnostics);
+    }
+
+    static ASR::expr_t *eval_ObjectType(Allocator &al, const Location &loc,
+            ASR::ttype_t* t1, Vec<ASR::expr_t*>& args) {
+        std::string object_type = "<type '" + ASRUtils::type_to_str_python(t1) + "'>";
+        return StringConstant(object_type, character(object_type.length()));
+    }
+
+    static inline ASR::asr_t* create_ObjectType(Allocator& al, const Location& loc,
+            Vec<ASR::expr_t*>& args,
+            const std::function<void (const std::string &, const Location &)> err) {
+        if (args.size() > 1) {
+            err("type() takes only 1 argument `object`", loc);
+        }
+
+        Vec<ASR::expr_t *> arg_values;
+        arg_values.reserve(al, 1);
+        arg_values.push_back(al, expr_value(args[0]));
```

Comment:
```suggestion
        Vec<ASR::expr_t *> arg_values;
```
We don't use the expr_value(args[0]) here, as the output is dependent on the argument type.